openzim/fcctozim/build.py

```diff
@@ -1,66 +1,95 @@
     with open(index_json_path) as course_index_str:
-        course_list = json.load(course_index_str)[language]
+        course_list = json.load(course_index_str)[fcc_lang]
 
     redirects = []
     for course in course_list:
-        meta_json_path = os.path.join(clientdir, 'src/assets/fcc/curriculum/', language, course, '_meta.json')
+        meta_json_path = os.path.join(
+            clientdir, "fcc/curriculum/", fcc_lang, course, "_meta.json"
+        )
         with open(meta_json_path) as meta_json_str:
-            challenges = json.load(meta_json_str)['challenges']
+            challenges = json.load(meta_json_str)["challenges"]
         for challenge in challenges:
-            redirects.append([ f'index.html#{language}/{course}/{challenge["slug"]}', challenge['title'] ])
-            
-    return redirects
+            title = challenge["title"]
+            redirects.append((f'{fcc_lang}/{course}/{challenge["slug"]}', title))
 
+    return OrderedDict(redirects).items()
 
-def build_zimfile(clientdir, outpath, language):
-    source_dir = os.path.join(clientdir, 'dist')
-    rootPath = os.path.join(source_dir, 'index.html')
+
+def build(arguments):
+    clientdir = arguments.clientdir
+    outpath = arguments.outzim
+    language = arguments.language
+    name = arguments.name
+    title = arguments.title
+    description = arguments.description
+
+    logger.info(
+        f"Building {clientdir} for {language} => {outpath} - Version: {VERSION}"
+    )
+
+    source_dir = os.path.join(clientdir)
+    rootPath = os.path.join(source_dir, "index.html")
     fileList = []
-    for (root, dirs, files) in os.walk(source_dir):
-        for name in files:
-            fileList.append(os.path.join(root, name))
 
-    fileList = [file for file in fileList if os.path.splitext(file)[1] != '.map']
+    # Walk the ree and get a list of files we care about
+    for root, dirs, files in os.walk(source_dir):
+        for name in files:
+            file = os.path.join(root, name)
+            if os.path.splitext(file)[1] == ".map":
+                continue
+            fileList.append(file)
 
     main_path = os.path.relpath(rootPath, source_dir)
+
+    # Make sure the outpath directory exists
+    outpathExists = os.path.exists(os.path.dirname(outpath))
+    if not outpathExists:
+        os.makedirs(os.path.dirname(outpath))
+
     tags = ";".join(["FCC", "freeCodeCamp"])
+
     with open(logo_path, "rb") as fh:
         png_data = fh.read()
+
     with Creator(outpath, main_path).config_dev_metadata(
-        Tags=tags, Illustration_48x48_at_1=png_data
+        Name=name,
+        Title=title,
+        Description=description,
+        Language=language,
+        Tags=tags,
+        Scraper=f"fcc2zim V{VERSION}",
+        Illustration_48x48_at_1=png_data,
     ) as creator:
         for file in fileList:
             print(file)
             path = os.path.relpath(file, source_dir)
             creator.add_item_for(path, fpath=file)
```

Comment:
All but the home page should be `is_front=False`

Reply:
When I do this, I can't search for content by title. Is there another way to handle this?